lib/public/utilities/fetch/buildUrl.js

```diff
@@ -74,22 +73,24 @@ export const buildUrl = (baseURL, parameters) => {
         }
     };
 
+    const [origin, existingParameters] = baseURL.split('?');
+
     if (existingParameters) {
         for (const formattedParameter of existingParameters.split('&')) {
-            const [key, value] = formattedParameter.split('=');
-            const [firstKey, ...dirtyKeys] = key.split('[');
-            const nestedKeys = [firstKey, ...dirtyKeys.map((key) => key.slice(0, -1))];
-
-            buildParameterFromNestedKeys(parameters, nestedKeys, value, null);
+            const parameterElements = formattedParameter.split(/\[|\]\[|\]/);
+            const nestedKeys = parameterElements.slice(0, parameterElements.length - 1);
+            const [lastKey, value] = parameterElements[parameterElements.length - 1].split('=');
+
+            if (lastKey === '') {
+                buildParameterFromNestedKeys(parameters, nestedKeys, value, null);
+            } else { // List nestedKeys is empty
+                buildParameterFromNestedKeys(parameters, [lastKey], value, null);
+            }
```

Comment:
Why this change?

Reply:
because, now you can put numeric operators as part of nested keys